doc/neps/dropping-python2.7-proposal.rst

```diff
@@ -0,0 +1,52 @@
+====================================
+Plan for dropping Python 2.7 support
+====================================
+
+The Python core team plans to stop supporting Python 2 in 2020. The NumPy
+project has supported both Python 2 and Python 3 in parallel since 2010, and
+has found that supporting Python 2 is an increasing burden on our limited
+resources; thus, we plan to eventually drop Python 2 support as well. Now that
+we're entering the final years of community-supported Python 2, the NumPy
+project wants to clarify our plans, with the goal of to helping our downstream
+ecosystem make plans and accomplish the transition with as little disruption as
+possible.
+
+Our current plan is as follows.
+
+Until **December 31, 2018**, all NumPy releases will fully support both
+Python2 and Python3.
+
+Starting on **January 1, 2019**, any new feature releases will support only
+Python3.
+
+The last Python2 supporting release will be designated as a long term support
+(LTS) release, meaning that we will continue to merge bug fixes and make bug
+fix releases for a longer period than usual.  Specifically, it will be
+supported by the community until **December 31, 2019**.
+
+On **January 1, 2020** we will raise a toast to Python2, and community support
+for the last Python2 supporting release will come to an end. However, it will
+continue to be available on PyPI indefinitely, and if any commercial vendors
+wish to extend the LTS support past this point then we are open to letting them
+use the LTS branch in the official NumPy repository to coordinate that.
+
+If you are a NumPy user who requires ongoing Python2 support in 2020 or later,
+then please contact your vendor. If you are a vendor who wishes to continue to
+support NumPy on Python2 in 2020+, please get in touch; ideally we'd like you
+to get involved in maintaining the LTS before it actually hits end of life so
+that we can make a clean handoff.
+
+To minimize disruption, running ``pip install numpy`` on Python 2 will continue
```

Comment:
Matthias Bussonier suggested to make it "should continue", have no clue myself, doubt it matters too much.

Reply:
2n's but yes :-) thanks. From experience with IPython many Python 2 users have an old verision of pip. Here is my [download statistics analysis](https://github.com/Carreau/talks/blob/master/2017-08-13-pybay/IPython-dls_2.ipynb) for IPython. Though with a good error message in setup.py (`please upgrade pip`) it tend to get solved fast.

Reply:
Maybe we should make that suggestion in the next release? It can't hurt to have it done before Python2 is dropped.

Reply:
For reference, the pip version should be >= 9.